VideoLocker/src/main/java/org/edx/mobile/util/ResourceUtil.java

```diff
@@ -38,4 +38,18 @@ public static CharSequence getFormattedStringForQuantity(@NonNull Resources reso
         String template = resources.getQuantityString(resourceId, quantity);
         return Phrase.from(template).put(key, quantity + "").format();
     }
+
+    public static CharSequence getFormattedString(@NonNull String pattern, @NonNull String key, @Nullable String value) {
+        return getFormattedString(pattern, Collections.singletonMap(key, value));
+    }
+
+    public static CharSequence getFormattedString(@NonNull String pattern, @NonNull Map<String, String> keyValMap) {
+        Phrase resourceString = Phrase.from(pattern);
+        Set<String> keys = keyValMap.keySet();
+        for (String key : keys) {
+            CharSequence val = keyValMap.get(key);
+            resourceString.put(key, val == null ? "" : val);
```

Comment:
Should we log when ``val`` is null? Seems like that would be unexpected and indicate a bug?